packages/expo-image-picker/ios/MediaHandler.swift

```diff
@@ -0,0 +1,347 @@
+  }
+
+  /**
+   @returns `true` upon copying success and `false` otherwise
+   */
+  static func tryCopyingOriginalImageFrom(mediaInfo: MediaInfo, to: URL) -> Bool {
+    guard let from = mediaInfo[.imageURL] as? URL else {
+      return false
+    }
+
+    do {
+      try FileManager.default.copyItem(atPath: from.path, toPath: to.path)
+      return true
+    } catch {
+      return false
+    }
+  }
+
+  /**
+   Reads base64 representation of the image data. If the data is `nil` fallbacks to reading the data from the url.
+   */
+  static func optionallyReadBase64From(
+    imageData: Data?,
+    orImageFileUrl url: URL,
+    tryReadingFile: Bool,
+    shouldReadBase64: Bool
+  ) throws -> String? {
+    if !shouldReadBase64 {
+      return nil
+    }
+
+    if tryReadingFile {
+      do {
+        let data = try Data(contentsOf: url)
+        return data.base64EncodedString()
+      } catch {
+        throw FailedToReadImageDataException()
+          .causedBy(error)
+      }
+    }
+
+    guard let data = imageData else {
+      throw FailedToReadImageDataForBase64Exception()
+    }
+
+    return data.base64EncodedString()
+  }
+
+  static func optionallyReadExifFrom(
+    mediaInfo: MediaInfo,
+    logger: EXLogManager?,
+    shouldReadExif: Bool,
+    completion: @escaping (_ result: [String: Any]?) -> Void
+  ) {
+    if !shouldReadExif {
+      return completion(nil)
+    }
+
+    let metadata = mediaInfo[.mediaMetadata] as? [String: Any]
+
+    if metadata != nil {
+      let exif = ImageUtils.readExifFrom(imageMetadata: metadata!)
+      return completion(exif)
+    }
+
+    guard let imageUrl = mediaInfo[.referenceURL] as? URL else {
+      logger?.info("Could not fetch metadata for image")
+      return completion(nil)
+    }
+
+    let assets = PHAsset.fetchAssets(withALAssetURLs: [imageUrl], options: nil)
+
+    guard let asset = assets.firstObject else {
+      logger?.info("Could not fetch metadata for image '\(imageUrl.absoluteString)'.")
+      return completion(nil)
+    }
+
+    let options = PHContentEditingInputRequestOptions()
+    options.isNetworkAccessAllowed = true
+    asset.requestContentEditingInput(with: options) { input, info in
+      guard let imageUrl = input?.fullSizeImageURL,
+            let properties = CIImage(contentsOf: imageUrl)?.properties
+      else {
+        logger?.info("Could not fetch metadata for '\(imageUrl.absoluteString)'.")
+        return completion(nil)
+      }
+      let exif = ImageUtils.readExifFrom(imageMetadata: properties)
+      return completion(exif)
+    }
+  }
+
+  static func readExifFrom(imageMetadata: [String: Any]) -> [String: Any] {
+    var exif: [String: Any] = imageMetadata[kCGImagePropertyExifDictionary as String] as? [String: Any] ?? [:]
+
+    // Copy ["{GPS}"]["<tag>"] to ["GPS<tag>"]
+    let gps = imageMetadata[kCGImagePropertyGPSDictionary as String] as? [String: Any]
+    if gps != nil {
+      gps!.forEach { key, value in
+        exif["GPD\(key)"] = value
```

Comment:
```suggestion
        exif["GPS\(key)"] = value
```
I'm not sure here but according to the comment, it should be GPS.

Reply:
Superb catch! 👏